src/ngraph/op/fused/grn.cpp

```diff
@@ -0,0 +1,90 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <iterator>
+
+#include "grn.hpp"
+#include "ngraph/axis_set.hpp"
+#include "ngraph/builder/norm.hpp"
+#include "ngraph/op/broadcast.hpp"
+#include "ngraph/op/divide.hpp"
+#include "ngraph/op/util/reshape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::GRN::GRN(const shared_ptr<Node>& data, float bias)
+    : FusedOp("GRN", {data})
+    , m_bias(bias)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::GRN::pre_validate_and_infer_types()
+{
+    const auto& data_pshape = get_input_partial_shape(0);
+
+    if (data_pshape.is_static())
+    {
+        const auto& data_shape{data_pshape.to_shape()};
+
+        // Input data must be 2, 3 or 4D tensor.
+        NODE_VALIDATION_CHECK(this,
+                              (data_shape.size() >= 2 && data_shape.size() <= 4),
+                              "Input tensor rank must be 2, 3 or 4 dimensional (actual input "
+                              "shape: ",
+                              data_shape,
+                              ").");
+    }
+}
+
+NodeVector op::GRN::decompose_op() const
+{
+    const auto input_node{get_argument(0)};
+    const auto& input_shape{input_node->get_shape()};
+
+    auto data{input_node};
```

Comment:
Can you operate on the `input_node` directly? I think you're making an unnecessary copy here.